docs/DeveloperGuide.md

```diff
@@ -1,38 +1,208 @@
 # Developer Guide
 
 ## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The following resources and libraries were referenced or adapted in this project:
+* Java Coding Standard (Basic): [Official Documentation](https://se-education.org/guides/conventions/java/basic.html)
+* Code Quality (Guide): (https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html)
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Main Components of the architecture
+
+The `Inventra` class manages the overall application flow, which includes application launch and shutdown.
+The core processes are managed by the following components:
+* **UI**: The user interface for handling user interaction and the display of output.
+* **Command**: The commands that user inputs.
+* **CommandParser**: A parser to handle said commands, as well as interpreting flags and arguments.
+* **Inventory**: The mode we are operating in.
+* **Storage**: Reads and writes data to a CSV file to maintain data persistence.
+
+### "Add" Command Feature
+The "AddCommand" is responsible for adding fields and records to the inventory. 
+Concurrently, "add" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+
+This command supports the following options:
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+
+#### Architecture Overview
+The 'AddCommand' is part of the `command` package, and interacts with the following key components:
+* **Inventory**: Contains the current state of the inventory, including fields and records.
+* **Csv**: Handles reading from and writing to the CSV file for data storage persistence.
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
```

Comment:
A "command flow" seems to suggest a sequence diagram more so than an architecture diagram.